### DISCUSSION TITLE
关于持续回话

### DISCUSSION THREAD

**Discussion by GTX-Yang** [simple-robot/Discussions#83] [Q&A]
### 涉及的编程语言

Java

### 涉及的组件库

mirai (QQ)

### 疑问描述

之前使用2.0开发的持续对话，现在在做升级。测试了一下发现没有会话key去控制了。请问有没有一个持续回话的例子。谢谢

**Comment by ForteScarlet** [comment]
https://simbot.forte.love/docs/basic/continuous-session

**Reply by GTX-Yang** [reply]
我知道有这个文档，看了，也写了代码。我的意思是2.0有key去控制上下文是否是同一个人；而3.0没有。所以造成了任何人回复都会触发。

**Reply by ForteScarlet** [reply]
直接给持续会话提供一个逻辑判断即可**避免任何人都会触发**的问题。

```java
context..waitingForNextMessage((context, event) -> {
        // 一些条件判断
        return "123456".equals(event.getAuthor().getId().toString());
    });
```

如果硬要说与2.0的key类似的东西，持续会话的第一个参数都有一个可选的 `id` 参数，比如 
```java
context.next("id", Event.Key)
```
